config/crawl.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const Crawler = require("simplecrawler");
+const chalk = require('chalk');
+
+const app = express();
+
+app.use(express.static(`${__dirname}/../_site`));
+
+// These pages incorporate content from other files in other repos, so
+// they should be considered "second class" by the link checker, and
+// only emit warnings on 404s rather than errors.
+const WARNING_PAGES = [
+    '/getting-started/code-guidelines/',
+    '/whats-new/releases/',
+    '/getting-started/showcase/all/',
+];
+const WARNING = chalk.yellow('Warning');
+const ERROR = chalk.red('Error');
+
+function shouldFetch(item, referrerItem) {
+  if (item.path.match(/&quot;/)) {
+    // If a URL's path contains a literal `&quot;` in it, then it's
+    // almost guaranteed to be a false-positive that's actually
+    // in an example snippet of HTML in the docs, so ignore it.
+    return false;
+  } else if (referrerItem.path.match(/\.js$/)) {
+    // Just ignore anything gleaned from JS files for now, it's too likely
+    // that it's a false positive.
+    return false;
+  }
+
+  return true;
+}
+
+const listener = app.listen(() => {
+  const port = listener.address().port;
+  const baseURL = `http://127.0.0.1:${port}`;
+  const crawler = new Crawler(`${baseURL}/`);
+  const referrers = {};
+  const notFound = [];
+
+  crawler.addFetchCondition((item, referrerItem, cb) => {
+    cb(null, shouldFetch(item, referrerItem));
+  });
+
+  crawler.maxDepth = 99;
+  crawler.interval = 1;
+  crawler.on("discoverycomplete", (item, resources) => {
+    resources.forEach(url => {
+      if (!(url in referrers)) {
+        referrers[url] = [];
+      }
+      referrers[url].push(item.path);
+    });
+  });
+  crawler.on("fetch404", (item, res) => {
+    notFound.push(item);
+  });
+  crawler.on("complete", () => {
+    listener.close(() => {
+      let errors = 0;
+      let warnings = 0;
+
+      notFound.forEach(item => {
+        const refs = referrers[item.url];
+        const isWarning = refs.every(path => WARNING_PAGES.includes(path));
+        const label = isWarning ? WARNING : ERROR;
+
+        console.log(`${label}: 404 for ${item.path}!`);
```

Comment:
leaving off the `!` would clean up the output a bit, and make it easier to copy the path